### ISSUE TITLE
Refactor the utils.go and marshaller/unmarshaller into separate packages

### ISSUE THREAD

**Issue by p4tin** [m110/watermill-amazonsqs#3]
Move those common components out of the `amazonsqs` package into a separate `utils` package?

**Comment by bkielbasa** [comment]
IMO `utils` package is a bad sign. It says nothing about what's there. You can call this file for example `endpoint.go`. WDYT?

**Comment by p4tin** [comment]
I can rename the file to endpoint.go but it is common between SQS and SNS so utils was going to be the name of the package and then the files can be name more appropriately does that sound better?

**Comment by m110** [comment]
I agree that `utils` sounds too generic, but first we need to decide if we want to keep SNS and SQS in separate packages? We could also keep all in `aws` package, but I'm not sure yet if that sounds reasonable.

**Comment by bkielbasa** [comment]
both queues are IMO completely different so IMO it sounds better to separate them. It simplifies the implementation.

**Comment by p4tin** [comment]
If you separate them completelty a lot of code would be duplicated... I would like to see a the Adapter be "aws" with "sqs" and "sns" packages. We could leave common structs and code in the "aws" package and the specific structs and code in the defined packages. This might also allow for the creation of a "s3" package in the future to deal with S3 file flows...  See PR #2  I think it's in that shape.
